### PR TITLE
Display linter messages that apply to the entire add-on

### PR DESCRIPTION
**Pull request by kumar303**
~~⚠️ Depends on https://github.com/mozilla/addons-code-manager/issues/884~~

Fixes https://github.com/mozilla/addons-code-manager/issues/805

As far as I can tell, these types of top-level messages [are rare](https://github.com/mozilla/addons-code-manager/issues/805#issuecomment-506776210). If we find something in the wild which produces many messages, we could revisit this UX.

Here are some examples with http://localhost:3000/en-US/browse/616915/versions/1688319/?path=background.js

Screenshot 2019-06-27 14 26 21
Screenshot 2019-06-27 14 26 45

### CODE REVIEW

**Comment by bobsilverberg** [comment]
I know you haven't asked for a review of this yet, but I was looking at in while reviewing #885. Can I suggest that you defer the changes to the signature of `createStoreWithVersion` to a separate PR because it makes what is otherwise a pretty small PR into a much larger one.

**Comment by kumar303** [comment]
> Can I suggest that you defer the changes to the signature of `createStoreWithVersion` to a separate PR because it makes what is otherwise a pretty small PR into a much larger one.

Really? The entire patch will only be a ~200 line change after https://github.com/mozilla/addons-code-manager/pull/885 lands. I could remove that part if it will help, though.

**Comment by bobsilverberg** [comment]
> > Can I suggest that you defer the changes to the signature of `createStoreWithVersion` to a separate PR because it makes what is otherwise a pretty small PR into a much larger one.
> 
> Really? The entire patch will only be a ~200 line change after #885 lands. I could remove that part if it will help, though.

I guess it's not that big of a deal. I just find it a bit distracting when there are a large number of files changed that don't really have anything to do with the feature. It's true that the number of lines changed isn't that big, but the number of files is, compared to what's needed for the issue. But sure, leave it be.

File: src/components/App/index.tsx [hunk 1]
```diff
@@ -152,14 +156,39 @@ export class AppBase extends React.Component<Props> {
...
           {this.renderErrors()}
+          {currentVersion && (
+            <LinterProvider
+              versionId={currentVersion.id}
+              validationURL={currentVersion.validationURL}
+              selectedPath={currentVersion.selectedPath}
+            >
+              {this.renderWithLinterProvider}
+            </LinterProvider>
+          )}
```
[the remark this answers is not in the document]
Comment (kumar303) [reply src/components/App/index.tsx]: > So, are we okay with leaking version stuff into the `App` component? I'd rather see this in the `Compare` and `Browse` page components or maybe into the `VersionFileViewer`.

I don't quite understand what you mean by leaking -- can you elaborate? In both the current patch (using `App`) and your proposed patch using `VersionFileViewer`, the data is mapped from a Redux store so what's the difference? I don't see what moving the mapping to `VersionFileViewer` solves. Did you mean to use `props.version` from `VersionFileViewer` instead?

Maybe it's because I don't consider the `App` component generic. It's a component specific to the shell of the code manager application. The main piece of data for all pages of code manager is the *current version* (as well as *current user*).

[the remark this answers is not in the document]
Comment (willdurand) [reply src/components/App/index.tsx]: > I don't quite understand what you mean by leaking -- can you elaborate? 

> Maybe it's because I don't consider the `App` component generic. It's a component specific to the shell of the code manager application. The main piece of data for all pages of code manager is the _current version_ (as well as _current user_).

> I don't see what moving the mapping to VersionFileViewer solves. 

Sorry. I mean putting code where it isn't supposed to be. The `App` component is the shell of all the page-level components. While having the current version in absolutely all pages _might_ make sense, I don't think having linter messages in all the pages does.

Example: in the near future, we might want to expose results of our various secret tools. Such results would (probably) be tied to a version and would likely be displayed on their own pages. We don't want to have (general) linter messages on these pages.

[the remark this answers is not in the document]
Comment (kumar303) [reply src/components/App/index.tsx]: ok, I see what you mean. That is a good reason for moving the logic to `VesionFileViewer` (the shared component for all version viewing). I will try moving it there. However, we probably won't get this separation 100% right because it's hard to predict the future.

File: stories/FileMetadata.stories.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/camelcase */
```
Comment (willdurand) [inline stories/FileMetadata.stories.tsx]: Why is this needed now? 🤔 (I see eslint complaining too)

Comment (kumar303) [reply stories/FileMetadata.stories.tsx]: There is some ghost in the machine but I haven't tried to track it down. My hunch is that the eslint rule gets applied in a later pass and there's a bug that relies on the order of loading files (or something like that).

File: src/components/FullscreenGrid/styles.module.scss [hunk 3]
```diff
@@ -26,17 +26,35 @@ $expand-button-width: 40px;
...
 }
 
+.Header,
+.topContent,
+.content,
+.mainSidePanel,
+.altSidePanel {
+  // Simulate a grid-gap.
+  margin-bottom: $default-padding;
+}
```
[the remark this answers is not in the document]
Comment (kumar303) [reply src/components/FullscreenGrid/styles.module.scss]: Yeah, I couldn't figure out how to keep `grid-gap`. It's a bit of a bummer because `grid-gap` is hard to polyfill in a robust way. For our case, we got lucky because we mostly have only full width columns -- only the middle three need horizontal gaps. It's very fragile now. If we change the column layout we'll have to remember to make gaps.

File: src/components/VersionFileViewer/index.spec.tsx [hunk 4]
```diff
@@ -88,7 +89,7 @@ describe(__filename, () => {
 
   type RenderParams = Partial<PublicProps>;
 
-  const render = (moreProps: RenderParams) => {
+  const renderRoot = (moreProps: RenderParams = {}) => {
```
[the remark this answers is not in the document]
Comment (kumar303) [reply src/components/VersionFileViewer/index.spec.tsx]: Good call, thanks. I was doing major surgery in here and had an old `renderWithLinterProvider` doing something different. I forgot I had deleted it so now I can use it again ✨

**Comment by kumar303** [comment]
> We should probably update the `FullscreenGrid` story to show an example with a top content. WDYT?

Good idea. Added in https://github.com/mozilla/addons-code-manager/pull/892/commits/3ba0121ac8f27a1ad5d7ef367b1ec6e50c5de34f
